src/dbally/views/sqlalchemy_base.py

```diff
@@ -63,16 +67,16 @@ async def _build_filter_bool_op(self, bool_op: syntax.BoolOp) -> sqlalchemy.Colu
 
     def execute(self, dry_run: bool = False) -> ExecutionResult:
         """
-        Executes the generated SQL query and returns the results.
-
-        :param dry_run: If True, only generate the query without executing it
+        Executes the generated SQL query and returns the results. Be aware that before running\
+        this method you need to execute `apply_filters`
 
-        :return: Query results
+        Args:
+            dry_run: If True, only fills the context field of the `ExecutionResult` with the query without executing it
```

Comment:
"only adds the SQL query to the context field without executing the query" (it doesn't really "fills the context field with the query")